gateway/src/main/java/com/lyft/data/gateway/router/RoutingManager.java

```diff
@@ -65,13 +66,29 @@ public void setBackendForQueryId(String queryId, String backend) {
   }
 
   /**
-   * Performs routing.
+   * Performs routing to an adhoc backend.
    *
    * @return
    */
-  public String provideBackendForThisRequest() {
```

Comment:
Consider a single provideBackendForThisRequest(boolean scheduled). Caller code in QueryIdCachingProxyHandler.java would be simplified as well.